apps/mobile-wallet/src/features/modals/BottomModal.tsx

```diff
@@ -18,249 +18,104 @@ along with the library. If not, see <http://www.gnu.org/licenses/>.
-
-    dispatch(removeModal({ id })) // Remove modal from stack after animation is done
-  }, [dispatch, id, onClose])
-
-  const handleClose = useCallback(() => {
-    'worklet'
-
-    navHeight.value = withSpring(0, springConfig)
-    modalHeight.value = withSpring(0, springConfig, (finished) => finished && runOnJS(handleCloseOnJS)())
-    position.value = 'closing'
-  }, [handleCloseOnJS, modalHeight, navHeight, position])
-
-  const handleMaximize = useCallback(() => {
-    'worklet'
-
-    navHeight.value = withSpring(NAV_HEIGHT + 6, springConfig)
-    modalHeight.value = withSpring(-maxHeight, springConfig)
-    position.value = 'maximised'
-  }, [maxHeight, modalHeight, navHeight, position])
-
-  const handleMinimize = useCallback(() => {
-    'worklet'
-
-    navHeight.value = withSpring(0, springConfig)
-    modalHeight.value = withSpring(-minHeight.value, springConfig)
-    position.value = 'minimised'
-  }, [minHeight.value, modalHeight, navHeight, position])
-
-  const panGesture = useMemo(
-    () =>
-      Gesture.Pan()
-        .onStart((e) => {
-          offsetY.value = modalHeight.value
-        })
-        .onChange((e) => {
-          if (position.value !== 'closing') {
-            modalHeight.value = offsetY.value + e.translationY
-          }
-        })
-        .onEnd(() => {
-          const shouldMinimise = position.value === 'maximised' && -modalHeight.value < dimensions.height - DRAG_BUFFER
-
-          const shouldMaximise =
-            canMaximize.value && position.value === 'minimised' && -modalHeight.value > minHeight.value + DRAG_BUFFER
-
-          const shouldClose =
-            ['minimised', 'closing'].includes(position.value) && -modalHeight.value < minHeight.value - DRAG_BUFFER
-
-          if (shouldMaximise) {
-            handleMaximize()
-          } else if (shouldMinimise) {
-            shouldMaximizeOnOpen.value ? handleClose() : handleMinimize()
-          } else if (shouldClose) {
-            handleClose()
-          } else {
-            modalHeight.value =
-              position.value === 'maximised'
-                ? withSpring(-maxHeight, springConfig)
-                : withSpring(-minHeight.value, springConfig)
-          }
-        }),
-    [
-      offsetY,
-      modalHeight,
-      position.value,
-      dimensions.height,
-      canMaximize.value,
-      minHeight.value,
-      handleMaximize,
-      shouldMaximizeOnOpen.value,
-      handleClose,
-      handleMinimize,
-      maxHeight
-    ]
-  )
-
-  // Trigger handle close when modal entity is closed from outside
-  useEffect(() => {
-    if (position.value !== 'closing' && modalEntity?.isClosing) {
-      handleClose()
-    }
-  }, [handleClose, modalEntity?.isClosing, position.value])
+  const {
+    modalHeightAnimatedStyle,
+    handleAnimatedStyle,
+    backdropAnimatedStyle,
+    handleContentSizeChange,
+    panGesture,
+    handleClose,
+    contentScrollHandlers,
+    isScrollable
+  } = useBottomModalState({
```

Comment:
Extracted the animation and gesture logic in a hook, allowing reusability in `BottomModalFlashList`.